third_party/proton/proton/proton.py

```diff
@@ -68,13 +76,29 @@ def run_profiling(args, target_args):
         import pytest
         pytest.main(script_args)
     else:
-        execute_as_main(script, script_args)
+        execute_as_main(script, script_args, instrumentation_pass)
+
+
+def run_profiling(args, target_args):
+    backend = args.backend if args.backend else _select_backend()
+
+    start(args.name, context=args.context, data=args.data, backend=backend, hook=args.hook)
+
+    do_setup_and_execute(target_args)
 
     finalize()
 
 
+def run_instrumentation(args, target_args):
+    backend = args.backend if args.backend else _select_backend()
+    do_setup_and_execute(target_args, args.instrument)
+
+
 def main():
     args, target_args = parse_arguments()
+    if args.instrument:
```

Comment:
I think it's worth mentioning that instrument and profiling are not compatible in command-line mode. We can use either functionality, but not both simultaneously.

Reply:
Updated in README, do you think we need a warning or a check here?

Reply:
I think it's fine